src/lib/firsts.rs

```diff
@@ -37,7 +37,7 @@ use cfgrammar::{Grammar, NTIdx, Symbol, TIdx};
 use cfgrammar::yacc::YaccGrammar;
 
 // Firsts stores all the first sets for a given grammar. For example, given this code and grammar:
```

Comment:
When I read this through I thought "Firsts" should be "First". In this comment I think your first line should start: "`Firsts` stores all the first sets..." to make it clear that this is a name from the code. The block of code underneath that shows how to create a grammar needs to be surrounded by triple-quotes.